src/app/dashboard/[interval]/[[...date]]/components/PullRequestsListModalContent.tsx

```diff
@@ -0,0 +1,44 @@
+import { ScrollArea } from "@/components/ui/scroll-area";
+import { ActivityItem } from "@/components/activity-item"; // This path needs to be checked/adjusted
+import { GitMerge, CircleDot } from "lucide-react";
+// Adjusted import path for IntervalMetrics
+import type { IntervalMetrics } from "../queries";
+
+
+interface PullRequestsListModalContentProps {
+  pullRequests: IntervalMetrics["topPullRequests"];
+}
+
+export default function PullRequestsListModalContent({
+  pullRequests,
+}: PullRequestsListModalContentProps) {
+  return (
+    <ScrollArea className="max-h-[80svh]">
+      <div className="divide-y">
+        {pullRequests.map((pr) => (
+          <ActivityItem
+            key={pr.id}
+            id={pr.id}
+            title={pr.title}
+            author={pr.author}
+            className="px-4"
+            number={pr.number}
+            href={`https://github.com/${pr.repository}/pull/${pr.number}`}
+            icon={
+              pr.mergedAt ? (
+                <GitMerge className="h-4 w-4 text-purple-500" />
+              ) : (
+                <CircleDot className="h-4 w-4 text-green-500" />
+              )
+            }
+          />
+        ))}
+        {pullRequests.length === 0 && (
+          <p className="p-4 text-center text-sm text-muted-foreground">
+            No pull requests to display.
+          </p>
+        )}
+      </div>
+    </ScrollArea>
```

Comment:
This also seems like its redefining components that already exist and aren't specific to dashboard page. e.g repeated logic for showing PR icons and constructing PR URLs. Not terrible, but ideally we reuse existing components where possible, or refactor them to be more reusable, instead of having many copies of similar code that will inevitably end up out of sync causing the UI/UX to be inconsistent and hard to maintain.